### PR TITLE
dev/lang: create a ruby page

### LINKED ISSUE #1116: Ethereum for Ruby developers
Don't we have a page for Ruby developers?

### PR DESCRIPTION
**Pull request by q9f**
## Description

Creates the page "Ethereum for Ruby Developers."

## Related Issue

fix #1116 #1300

### CODE REVIEW

File: src/content/developers/docs/programming-languages/ruby/index.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+title: Ethereum for Ruby Developers
+description: Learn how to develop for Ethereum using Ruby-based projects and tooling.
+lang: en
+sidebar: true
+incomplete: true
+---
+
+<div class="featured">Learn how to develop for Ethereum using Ruby-based projects and tooling.</div>
+
+Use Ethereum to create decentralized applications (or "dapps") that utilize the benefits of cryptocurrency and blockchain technology. These dapps can be trustworthy, meaning that once they are deployed to Ethereum, they will always run as programmed. They can control digital assets to create new kinds of financial applications. They can be decentralized, meaning that no single entity or person controls them and are nearly impossible to censor.
```
Comment (nassersaazi) [inline src/content/developers/docs/programming-languages/ruby/index.md:11]:
```suggestion
Use Ethereum to create decentralized applications (or "dapps") that utilize the benefits of cryptocurrency and blockchain technology. These dapps can be trustless, meaning that once they are deployed to Ethereum, they will always run as programmed. They can control digital assets to create new kinds of financial applications. They can be decentralized, meaning that no single entity or person controls them and are nearly impossible to censor.
```

File: src/content/developers/docs/programming-languages/ruby/index.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+- [How to generate a new Ethereum address in Ruby](https://www.quicknode.com/guides/web3-sdks/how-to-generate-a-new-ethereum-address-in-ruby)
+
+## Intermediate articles {#intermediate-articles}
+
+- [Blockchain App with Ruby](https://www.nopio.com/blog/blockchain-app-ruby/)
+- [Use the Ruby, connected to Ethereum, to execute the Smart Contract](https://titanwolf.org/Network/Articles/Article?AID=87285822-9b25-49d5-ba2a-7ad95fff7ef9)
+
+## Ruby projects and tools {#ruby-projects-and-tools}
+
+### Active: {#active}
```
Comment (minimalsm) [inline src/content/developers/docs/programming-languages/ruby/index.md:37]: I know this template was copied from another programming language but we shouldn't have colons at the end of headings.

Screenshot 2022-02-22 at 11 11 54

```suggestion
### Active {#active}
```

File: src/content/developers/docs/programming-languages/ruby/index.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+- [eth.rb](https://github.com/q9f/eth.rb) - _Ruby library and RPC-client to handle Ethereum accounts, messages, and transactions_
+- [siwe-ruby](https://github.com/spruceid/siwe-ruby) - _Ruby implementation of Sign-In with Ethereum_
+- [keccak.rb](https://github.com/q9f/keccak.rb) - _The Keccak (SHA3) hash used by Ethereum_
+- [ethereum-on-rails](https://github.com/q9f/ethereum-on-rails) - _Ethereum on Rails template which allows to connect MetaMask to Ruby on Rails_
+- [omniauth-nft](https://github.com/valthon/omniauth-nft) - _OmniAuth strategy for authenticating via NFT ownership._
+- [omniauth-ethereum.rb](https://github.com/q9f/omniauth-ethereum.rb) - _Implements the Ethereum provider strategy for OmniAuth._
+
+### Archived / No longer maintained: {#archived--no-longer-maintained}
```
Comment (minimalsm) [inline src/content/developers/docs/programming-languages/ruby/index.md:46]:
```suggestion
### Archived / No longer maintained {#archived--no-longer-maintained}
```

File: src/content/developers/docs/programming-languages/ruby/index.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+- [web3-eth](https://github.com/spikewilliams/vtada-ethereum) - _Calling RPC methods of Ethereum node with Ruby_
+- [ethereum_tree](https://github.com/longhoangwkm/ethereum_tree) - _Ruby library to generate ETH addresses from a Hierarchical Deterministic wallet according to the BIP32 standard_
+- [etherlite](https://github.com/budacom/etherlite) - _Ethereum integration for Ruby on Rails._
+- [ethereum.rb](https://github.com/EthWorks/ethereum.rb) - _Ruby Ethereum client using the JSON-RPC interface for sending transactions, creating and interacting with contracts as well as usefull toolkit to work with Ethereum node._
+
+Looking for more resources? Check out [ethereum.org/developers](/developers/).
```
Comment (minimalsm) [inline src/content/developers/docs/programming-languages/ruby/index.md:53]:
```suggestion
Looking for more resources? Check out [our Developers' home](/developers/).
```

File: src/content/developers/docs/programming-languages/ruby/index.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+## Ruby Community Contributors {#ruby-community-contributors}
```
Comment (minimalsm) [inline src/content/developers/docs/programming-languages/ruby/index.md:55]:
```suggestion
## Ruby community contributors {#ruby-community-contributors}
```

File: src/content/developers/docs/programming-languages/ruby/index.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+The [Ethereum Ruby Telegram Group](https://t.me/ruby_eth) is a host to a rapidly growing community and is the dedicated resource for discussions on any of the above projects and related topics.
```
Comment (minimalsm) [inline src/content/developers/docs/programming-languages/ruby/index.md:57]:
```suggestion
The [Ethereum Ruby Telegram group](https://t.me/ruby_eth) is a host to a rapidly growing community and is the dedicated resource for discussions on any of the above projects and related topics.
```

**Review by minimalsm — CHANGES_REQUESTED** [review]
Thanks for creating this @q9f ❤️❤️❤️❤️ noticed a few problems with our template that I've suggested changes for but nothing major.

Could you add this to the [developer-docs-links YAML](https://github.com/ethereum/ethereum-org-website/blob/27236e8bcb6203a92bb881772923900d594994f1/src/data/developer-docs-links.yaml#L138) file to that it appears in the side navigation?

File: src/content/developers/docs/programming-languages/ruby/index.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
+---
+title: Ethereum for Ruby Developers
+description: Learn how to develop for Ethereum using Ruby-based projects and tooling.
+lang: en
+sidebar: true
+incomplete: true
```
Comment (minimalsm) [inline src/content/developers/docs/programming-languages/ruby/index.md:6]: Also, you've marked this as incomplete in the frontmatter. This displays a banner at the top of the page and a call to action at the bottom of the page. I'll default to you as the expert on the Ruby Ethereum ecosystem here — would you say it is accurate to call this page incomplete?

Screenshot 2022-02-22 at 11 33 12

Screenshot 2022-02-22 at 11 34 29

**Comment by minimalsm** [comment]
@q9f just pinging to make sure you've seen this 🙏

File: src/content/developers/docs/programming-languages/ruby/index.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+Use Ethereum to create decentralized applications (or "dapps") that utilize the benefits of cryptocurrency and blockchain technology. These dapps can be trustworthy, meaning that once they are deployed to Ethereum, they will always run as programmed. They can control digital assets to create new kinds of financial applications. They can be decentralized, meaning that no single entity or person controls them and are nearly impossible to censor.
+
+## Getting started with smart contracts and the Solidity language {#getting-started-with-smart-contracts-and-solidity}
+
+**Take your first steps to integrating Ruby with Ethereum**
+
+Need a more basic primer first? Check out [ethereum.org/learn](/learn/) or [ethereum.org/developers](/developers/).
+
+- [Blockchain Explained](https://kauri.io/article/d55684513211466da7f8cc03987607d5/blockchain-explained)
```
Comment (minimalsm) [inline src/content/developers/docs/programming-languages/ruby/index.md:19]: Let's link internally here

**Comment by q9f** [comment]
> @q9f just pinging to make sure you've seen this pray

Thank you for the reviews! I'll address them by end of the week!

File: src/content/developers/docs/programming-languages/ruby/index.md (hunk 8 above)
Comment (minimalsm) [reply src/content/developers/docs/programming-languages/ruby/index.md:19]: No need to do anything here @q9f, this was just a note to myself.

File: src/content/developers/docs/programming-languages/ruby/index.md (hunk 7 above)
Comment (q9f) [reply src/content/developers/docs/programming-languages/ruby/index.md:6]: No, this page is fairly complete now. :-)

**Review by minimalsm — APPROVED** [review]
Merging @q9f, thanks again 🎉🎉🎉🎉

Be sure to claim your [2022 POAPs](https://ethereum.org/en/contributing/#poap) also 🥳!

**Comment by minimalsm** [comment]
@all-contributors please add @q9f for code, docs

**Pull request merged**